kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQuerySinkConnectorIT.java

```diff
@@ -141,86 +147,112 @@
+  private final List<List<Object>> expectedRows;
+  private final String topic;
+  private final String table;
+  private final String connectorName;
 
-  private RestApp restApp;
-  private String schemaRegistryUrl;
   private Producer<byte[], byte[]> valueProducer;
   private int numRecordsProduced;
 
-  @Before
-  public void setup() throws Exception {
-    Collection<String> tables = TEST_TOPICS.stream()
-        .map(this::suffixedAndSanitizedTable)
-        .collect(Collectors.toSet());
-    BucketClearer.clearBucket(keyFile(), project(), gcsBucket(), gcsFolder(), keySource());
-    TableClearer.clearTables(newBigQuery(), dataset(), tables);
-
-    startConnect();
-    restApp = new RestApp(
+  public BigQuerySinkConnectorIT(String testCase, List<List<Object>> expectedRows) {
+    this.testCase = testCase;
+    this.expectedRows = expectedRows;
+
+    this.topic = TEST_CASE_PREFIX + testCase;
+    this.table = testBase.suffixedAndSanitizedTable(topic);
+    this.connectorName = "bigquery-connector-" + testCase;
+  }
+
+  @BeforeClass
+  public static void globalSetup() throws Exception {
+    testBase = new BaseConnectorIT() {};
+    testBase.startConnect();
+
+    schemaRegistry = new RestApp(
         ClusterTestHarness.choosePort(),
         null,
-        connect.kafka().bootstrapServers(),
+        testBase.connect.kafka().bootstrapServers(),
         SchemaRegistryConfig.DEFAULT_KAFKASTORE_TOPIC,
         CompatibilityLevel.BACKWARD.name,
         true,
         null);
 
-    restApp.start();
+    schemaRegistry.start();
 
-    schemaRegistryUrl = restApp.restClient.getBaseUrls().current();
+    schemaRegistryUrl = schemaRegistry.restClient.getBaseUrls().current();
+
+    BucketClearer.clearBucket(
+      testBase.keyFile(),
+      testBase.project(),
+      testBase.gcsBucket(),
+      testBase.gcsFolder(),
+      testBase.keySource()
+    );
+  }
+
+  @Before
+  public void setup() {
+    TableClearer.clearTables(testBase.newBigQuery(), testBase.dataset(), table);
 
     Map<String, Object> producerProps = new HashMap<>();
-    producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, connect.kafka().bootstrapServers());
+    producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, testBase.connect.kafka().bootstrapServers());
     valueProducer = new KafkaProducer<>(
         producerProps, Serdes.ByteArray().serializer(), Serdes.ByteArray().serializer());
 
     numRecordsProduced = 0;
   }
 
   @After
-  public void close() throws Exception {
-    if (restApp != null) {
-      restApp.stop();
+  public void cleanup() {
+    testBase.connect.deleteConnector(connectorName);
+  }
+
+  @AfterClass
+  public static void globalCleanup() {
+    if (schemaRegistry != null) {
+      Utils.closeQuietly(schemaRegistry::stop, "embedded Schema Registry instance");
     }
-    stopConnect();
+    testBase.stopConnect();
   }
 
   @Test
-  public void testAll() throws Exception {
+  public void runTestCase() throws Exception {
```

Comment:
I feel like we're not appropriately taking advantage of the parameterized test by only having one case with a generic name. This is almost golang-style tests. I think that this might just be a lateral change until we have multiple cases to test here (multiple schema variants, or whatever else).

Reply:
> I think that this might just be a lateral change until we have multiple cases to test here (multiple schema variants, or whatever else).

I disagree; as mentioned in the description: "Each test case now uses a different connector (to make it possible to run several test cases in a single build and get a more granular signal of success/failure for each of them)". We can now identify several failing test cases in a single run instead of just seeing one failing test case.

Reply:
I think that the increased granularity is worth it.